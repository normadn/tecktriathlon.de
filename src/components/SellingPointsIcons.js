import React from "react";

const SellingPointsIcons = (props) => (
  <div
    css={{
      display: "grid",
      gridTemplateColumns: "1fr ".repeat(props.list.length),
      width: "calc(100% - 200px)",
      margin: "0 auto",
      minWidth: "300px"
    }}
  >
    {props.list.map((item) => (
      <div key={item.text}>
        <img src={item.icon} alt="" />
        <div
          css={{
            fontWeight: 600,
            fontSize: "24px",
            padding: "6px 0"
          }}
        >
          {item.number}+
        </div>
        <div
          css={{
            fontWeight: 500,
            fontSize: "14px"
          }}
        >
          {item.text}
        </div>
      </div>
    ))}
  </div>
);

export default SellingPointsIcons;
