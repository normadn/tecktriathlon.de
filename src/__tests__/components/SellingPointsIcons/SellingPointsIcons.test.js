import React from "react";
import { shallow } from "enzyme";
import SellingPointsIcons from "../../../components/SellingPointsIcons";
import hackers from "../assets/img/hackers.svg";
import schools from "../assets/img/schools.svg";
import applications from "../assets/img/applications.svg";

describe("SellingPointsIcons", () => {
  it("renders correctly", () => {
    expect(
      shallow(
        <SellingPointsIcons
          list={[
            {
              icon: applications,
              number: 1800,
              text: "Applications"
            },
            {
              icon: schools,
              number: 25,
              text: "Schools"
            },
            {
              icon: hackers,
              number: 500,
              text: "Hackers"
            }
          ]}
        />
      )
    ).toMatchSnapshot();
  });
});
