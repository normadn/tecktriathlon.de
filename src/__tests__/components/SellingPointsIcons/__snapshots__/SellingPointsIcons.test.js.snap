// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellingPointsIcons renders correctly 1`] = `
<div
  css={
    Object {
      "display": "grid",
      "gridTemplateColumns": "1fr 1fr 1fr ",
      "margin": "0 auto",
      "minWidth": "300px",
      "width": "calc(100% - 200px)",
    }
  }
>
  <div
    key="Applications"
  >
    <img
      alt=""
      src="test-file-stub"
    />
    <div
      css={
        Object {
          "fontSize": "24px",
          "fontWeight": 600,
          "padding": "6px 0",
        }
      }
    >
      1800
      +
    </div>
    <div
      css={
        Object {
          "fontSize": "14px",
          "fontWeight": 500,
        }
      }
    >
      Applications
    </div>
  </div>
  <div
    key="Schools"
  >
    <img
      alt=""
      src="test-file-stub"
    />
    <div
      css={
        Object {
          "fontSize": "24px",
          "fontWeight": 600,
          "padding": "6px 0",
        }
      }
    >
      25
      +
    </div>
    <div
      css={
        Object {
          "fontSize": "14px",
          "fontWeight": 500,
        }
      }
    >
      Schools
    </div>
  </div>
  <div
    key="Hackers"
  >
    <img
      alt=""
      src="test-file-stub"
    />
    <div
      css={
        Object {
          "fontSize": "24px",
          "fontWeight": 600,
          "padding": "6px 0",
        }
      }
    >
      500
      +
    </div>
    <div
      css={
        Object {
          "fontSize": "14px",
          "fontWeight": 500,
        }
      }
    >
      Hackers
    </div>
  </div>
</div>
`;
